src/main/scala/strawman/collection/Iterable.scala

```diff
@@ -81,17 +81,21 @@ trait IterableOps[+A] extends Any {
   /** A view representing the elements of this collection. */
   def view: View[A] = View.fromIterator(iterator())
 
-  /** Given a collection factory `fi` for collections of type constructor `C`,
-    *  convert this collection to one of type `C[A]`. Example uses:
+  /** Given a collection factory `fi`, convert this collection to the appropriate
+    * representation for the current element type `A`. Example uses:
     *
     *      xs.to(List)
     *      xs.to(ArrayBuffer)
+    *      xs.to(BitSet) // for xs: Iterable[Int]
     */
-  def to[C[X] <: Iterable[X]](fi: FromIterable[C]): C[A @uncheckedVariance] =
+  def to[F <: FromIterableBase[A]](fi: F): F#To[A @uncheckedVariance] =
   // variance seems sound because `to` could just as well have been added
   // as a decorator. We should investigate this further to be sure.
     fi.fromIterable(coll)
 
+  def to[C[_], Ev[_]](fi: ConstrainedFromIterable[C, Ev])(implicit ev: Ev[A @uncheckedVariance]): C[A @uncheckedVariance] =
+    fi.constrainedFromIterable(coll)
```

Comment:
Right, this still needs to be improved. The unconstrained version cannot work for binary type constructors. We only have an element type `A` in `Iterable`. While we can restrict it to work only for `(A1,A2)` there is no way to compute these types `A1` and `A2` without implicits. Conversions to maps either need another special case or a `ConstrainedFromIterable`. Since we're special-casing `Map` in other places we should probably overload `to` with two additional cases for unconstrained and constrained maps.